flake.nix

```
@@ -49,12 +49,22 @@
                   })).config;
 
                 moduleDeclarationFile =
-                  (builtins.unsafeGetAttrPos "modules" args).file;
+                  let
+                    # Even though `modules` is a mandatory argument for `nixosSystem`, it doesn't
+                    # mean that the evaluator keeps track of its position. If there are too many levels
+                    # of indirection, the position gets lost at some point and `EvalState::mkPos`
+                    # returns `null` in `libexpr`.
```

```suggestion
                    # of indirection, the position gets lost at some point.
```

Too much detail (I think)